### PR TITLE
Provider/requestor switch

### PR DESCRIPTION
**Pull request by Wiezzel**
* Stop accepting tasks when a Task API task is requested.
* Resume accepting tasks when no more Task API tasks are requested.

### CODE REVIEW

File: golem/task/requestedtaskmanager.py [hunk 1]
```diff
@@ -235,6 +236,12 @@ def is_task_finished(task_id: TaskId) -> bool:
         task = RequestedTask.get(RequestedTask.task_id == task_id)
         return task.status.is_completed()
 
+    @staticmethod
+    def has_unfinished_tasks() -> bool:
+        """ Return True iff there are any tasks that need computation. """
```
Comment (maaktweluit) [inline golem/task/requestedtaskmanager.py:241]:
```suggestion
        """ Return True if there are any tasks that need computation. """
```

**Review by maaktweluit — APPROVED** [review]
LGTM!

One small comment, still approving!

File: golem/task/requestedtaskmanager.py (hunk 1 above)
Comment (Wiezzel) [reply golem/task/requestedtaskmanager.py:241]: 'iff' is a shorthand for 'if and only if'

Comment (maaktweluit) [reply golem/task/requestedtaskmanager.py:241]: ok, TIL ^^

**Comment by mfranciszkiewicz** [comment]
There is another side of the issue, unresolved - we're still able to request tasks while computing a subtask. But I’m fine with doing so separately

**Comment by Wiezzel** [comment]
> There is another side of the issue, unresolved - we're still able to request tasks while computing a subtask. But I’m fine with doing so separately

Take a look at https://github.com/golemfactory/golem/pull/4837/commits/4d97349c05a84b79454819d91bc070ceda5d9e17

**Review by mfranciszkiewicz — APPROVED** [review]

**Pull request merged**